routing/index_graph.hpp

```diff
@@ -29,23 +32,129 @@ class JointEdge final
 
 private:
   // Target is vertex going to for outgoing edges, vertex going from for ingoing edges.
-  Joint::Id const m_target;
-  double const m_weight;
+  Joint::Id m_target;
+  double m_weight;
+};
+
+class JointEdgeGeom final
+{
+public:
+  JointEdgeGeom() = default;
+  JointEdgeGeom(Joint::Id target, vector<RoadPoint> const & path) : m_target(target), m_path(path)
+  {
+  }
+  Joint::Id GetTarget() const { return m_target; }
+  vector<RoadPoint> const & GetPath() const { return m_path; }
+
+private:
+  // Target is vertex going to for outgoing edges, vertex going from for ingoing edges.
+  Joint::Id m_target = Joint::kInvalidId;
+  vector<RoadPoint> m_path;
+};
+
+class DirectedEdge final
```

Comment:
BTW, why do we need JointEdge class at all? It seems that these two classes do almost the same, right? Moreover, from my point of view this class may completely replace JointEdge class.